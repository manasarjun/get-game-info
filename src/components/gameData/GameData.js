import React, { useState } from "react";
import PropTypes from 'prop-types';

import StartsInfo from '../StartsInfo/StartsInfo';

function GameData({ data }) {

  const [isCollapsable, setIsCollapsable] = useState(false);
  const [currentIndex, setCurrentIndex] = useState(null);

  const handleShow = (index) => () => {
    setIsCollapsable(true);
    setCurrentIndex(index);
  }

  const handleCollapsable = () => () => {
    setIsCollapsable(false)
  }

  const prettyPrint = (string) => {
    const t = string.indexOf('T');
    const date = string.substring(0, t);
    const time = string.substring(t + 1);
    return <>{date} <b>{time}</b></>;
  };

  return (
    <div className="data-container">

      <ul className='game-data'>
        <h3>Races</h3>
        {data && data.races.map(
          (race, index) => (
            <li key={index} className='data-items'>
              <p>Race Number: {race.number}</p>
              {race.name && <p>Race Name: {race.name}</p>}
              <p>Race Start Time: {prettyPrint(race.startTime)}</p>
              {
                (currentIndex === index && isCollapsable) ?
                  <>
                    <button onClick={handleCollapsable(setIsCollapsable)}>
                      Hide StartsInfo
                      </button>
                    <StartsInfo starts={race.starts} />
                  </>
                  :
                  <button onClick={handleShow(index, setCurrentIndex, setIsCollapsable)}>
                    Show StartsInfo
                    </button>
              }
            </li>)
        )}
      </ul>
    </div >
  );
}

GameData.propTypes = {
  data: PropTypes.shape({
    number: PropTypes.number,
    name: PropTypes.string,
    startTime: PropTypes.string,
    starts: PropTypes.array
  })
}

export default GameData;
