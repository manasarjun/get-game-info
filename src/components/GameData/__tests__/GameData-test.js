import React from 'react';
import renderer from 'react-test-renderer';
//import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import GameData from '../GameData'
Enzyme.configure({ adapter: new Adapter() });

describe.only('<GameData />', () => {

  const baseProps = {
    races: [
      {
        name: "Breddlopp - SpÃ¥rtrappa",
        number: 8,
        startTime: "2020-11-14T21:02:00",
        starts: [{}, {}],
      },
      {

      }

    ],
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<GameData {...baseProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });



});
