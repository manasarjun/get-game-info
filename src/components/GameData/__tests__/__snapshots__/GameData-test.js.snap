// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameData /> renders correctly 1`] = `
<div
  className="data-container"
>
  <ul
    className="game-data"
  >
    <h3>
      Races
    </h3>
    <li
      className="data-items"
      key="0"
    >
      <p>
        Race Number: 
        8
      </p>
      <p>
        Race Name: 
        Breddlopp - Sp√•rtrappa
      </p>
      <p>
        Race Start Time: 
        2020-11-14
         
        <b>
          21:02:00
        </b>
      </p>
      <button
        onClick={[Function]}
      >
        Show StartsInfo
      </button>
    </li>
  </ul>
</div>
`;
